api/serializers/__init__.py

```diff
@@ -105,7 +109,39 @@ class Meta:
         ]
 
 
-class ProfileSerializer(serializers.ModelSerializer):
+class StrictReadOnlyFieldsMixin:
```

Comment:
doc (blocker): add a link from this code to https://github.com/encode/django-rest-framework/issues/1655#issuecomment-1197033853 where it came from. So future us's know where this code came from.

Reply:
suggestion (non-blocking): Also, the work-around in [this comment](https://github.com/encode/django-rest-framework/issues/1655#issuecomment-1478741260) is less code. I'm not sure if it's better code, but did you consider it too?

Reply:
> suggestion (non-blocking): Also, the work-around in [this comment](https://github.com/encode/django-rest-framework/issues/1655#issuecomment-1478741260) is less code. I'm not sure if it's better code, but did you consider it too?

With the new mixin class, all of the read-only fields give a 400 response on PUT/PATCH requests. I think this makes sense because we don't want any of them giving a 200 response. And this class would also be reusable for the other serializers with read-only fields. 

However, if we want just the subdomain field to give a 400 code, I think the latter code ([this comment](https://github.com/encode/django-rest-framework/issues/1655#issuecomment-1478741260)) is better. 

@groovecoder let me know if we should make all of the read-only fields respond with 400 or just the subdomain.  Will also make sure to add this into the docs. Thanks!

Reply:
> With the new mixin class, all of the read-only fields give a 400 response on PUT/PATCH requests ... And this class would also be reusable for the other serializers with read-only fields.

Sounds good! Let's keep the mixin.